src/vfs/public.c

```diff
@@ -152,6 +161,36 @@ static bool vfs_mkparents_recurse(const char *path) {
 
 	return vfs_mkparents_recurse(p) && vfs_mkdir(p);
 }
+*/
+static bool vfs_mkparents_recurse(const char *path) {
+
+    char p[strlen(path) + 1];
+    VFSInfo i;
+    strcpy(p,path);
+    while(*p){
+        char *psep = strrchr(p, VFS_PATH_SEPARATOR);
+
+        if(!psep){
+            //current parent is root
+            return true;
+        }
+
+        *psep = 0;
+
+        i = vfs_query(p);
+
+        if(i.error){
+            return false;
+        }else if(i.exists){
+            return true;
```

Comment:
If it already exists, then we should only return `true` if it's a directory (`i.is_dir`); otherwise that's an error condition (return false).